src/Castle.Core.Tests/DynamicProxy.Tests/ComInterfacesTests.cs

```diff
@@ -35,26 +35,47 @@ public void Marshal_Release_throws_when_called_with_IntPtr_Zero()
 			// See https://github.com/mono/mono/issues/15853.
 		}
 
+		// NOTE: You may wonder why the following tests delegate their implementation to static methods
+		// in a private static class. This is because the method bodies contain references to ADODB.dll,
+		// which fails to load under Mono/Linux, causing test discovery to abort early. By moving these
+		// method bodies somewhere else, test discovery will succeed and we get a chance at specifying
+		// how the tests should run, e.g. using the `[Platform]` attributes.
+
 		[Test]
+		[Platform(Include = "Win", Reason = "Depends on ADODB.dll, which can fail to load under Mono/Linux.")]
```

Comment:
To be honest, I am being lazy here: NUnit's `[Platform]` is not available on .NET Standard / Core, so this compiles only because these targets are currently excluded by `FEATURE_TEST_COM`. It would be more correct to e.g. declare a test `[Category("Windows")]`, and then exclude that category from the test run in `build.sh`. Or build a `[IncludeOnOS]` attribute similar to our `[ExcludeOnFramework]`.

Reply:
Happy for it to be this way for now. We've actually got a bunch of this sort of thing where it only works because of the conditional compilation of something else, we'll consolidate when we remove .NET 3.5 and .NET Standard 1.x support in the future.